plugins/feeds/public/et_open.py

```diff
@@ -0,0 +1,138 @@
+            return
+
+        if self.__filter_rule(rule_suricata.metadata):
+            ind_suricator = indicator.Suricata(
+                name=rule_suricata["msg"],
+                pattern=line,
+                metadata=rule_suricata.metadata,
+                diamond=indicator.DiamondModel.infrastructure,
+                sid=rule_suricata["sid"],
+            ).save()
+
+            if "cve" in ",".join(rule_suricata.metadata):
+                for ind_cve in self.__extract_cve(rule_suricata.metadata):
+                    ind_suricator.link_to(ind_cve, "affect", "ETOpen")
+
+            if "malware family" in ",".join(rule_suricata.metadata):
+                for in_malware_family in self.__extract_malware_family(
+                    rule_suricata.metadata
+                ):
+                    ind_suricator.link_to(in_malware_family, "affect", "ETOpen")
+
+            tags = self.__extract_tags(rule_suricata.metadata)
+            if tags:
+                ind_suricator.tag(tags)
+            if "mitre_tactic_id" in ",".join(rule_suricata.metadata):
+                for ind_mitre_attack in self.__extract_mitre_attack(
+                    rule_suricata.metadata
+                ):
+                    if ind_mitre_attack:
+                        ind_suricator.link_to(ind_mitre_attack, "affect", "ETOpen")
+
+    def __extract_cve(self, metadata: list):
+        for meta in metadata:
+            if meta.startswith("cve"):
+                _, cve = meta.split(" ")
+                if "_" in cve:
+                    cve = cve.replace("_", "-")
+                ind_cve = entity.Vulnerability.find(name=cve)
+                if not ind_cve:
+                    ind_cve = entity.Vulnerability(name=cve).save()
+                yield ind_cve
+
+    def __extract_malware_family(self, metadata: list):
+        for meta in metadata:
+            if "malware family" in meta:
+                _, malware_family = meta.split(" ")
+                ind_malware_family = entity.Malware.find(name=malware_family)
+                if not ind_malware_family:
+                    ind_malware_family = entity.Malware(name=malware_family).save()
+                yield ind_malware_family
+
+    def __extract_tags(self, metadata: list):
+        tags = []
+        for meta in metadata:
+            if meta.startswith("tag"):
+                _, tag = meta.split(" ")
+                tags.append(tag)
+        return tags
+
+    def __extract_mitre_attack(self, metadata: list):
+        for meta in metadata:
+            if "mitre_tactic_id" in meta:
+                _, mitre_id = meta.split(" ")
+                ind_mitre_attack, nb_ent = entity.Entity.filter(
+                    query_args={"type": entity.EntityType.attack_pattern},
+                    aliases=mitre_id,
+                )
+                if nb_ent != 0:
+                    yield ind_mitre_attack[0]
+
+    def __filter_rule(self, metadata):
+        if not self.last_run:
```

Comment:
It would be nice to add a docstring that explains what this function is doing at a high level